### DISCUSSION TITLE
Videos keep resizing when moving from one monitor to another

### DISCUSSION THREAD

**Discussion by dmvmoe** [iina/iina#4240] [General]
I'm not sure what this is about.  Mac Min, M1, 2020, Venture 13.2.1, IINA 1.3.1.  When I open a video on my left screen it's one size.  When I move it over to the right size, it's another size.  Both are on 2560 x 1140 as the default and at 60 Hertz.  I can't for the life of me figure out what is going on. This only happens videos.  Any suggestions.  I've changed to many settings to count so I've uninstalled and re-installed have touched nothing.  Videos still resize when moving video so the other monitor.  What gives?

**Comment by svobs** [comment]
I'm currently investigating IINA's window resizing code, and...there are a lot of issues.

By any chance, is one of your displays a retina display and the other a non-retina display? It looks like moving the window between a retina screen to a non-retina screen isn't properly handled, and results in the video size being reduced by half, doubled, or something else.

Also, did you resize the window in any way before moving it between screens? This includes using the items in the `Video` menu. I'm sometimes seeing an issue where IINA isn't getting a response from its mpv backend and therefore doesn't update its scale factor, even though it's already updated the window size...which ends up causing the next set of window calculations to be incorrect. However, it's proving difficult to reproduce reliably.

**Comment by ballo** [comment]
Related to #4424?
